# Intro to Lists

List adalah kumpulan elemen yang disusun secara terurut dan dapat diubah (mutable). Setiap elemen dalam list memiliki posisi yang ditentukan oleh indeks numerik, dimulai dari 0.

In [127]:
flowers = "pink primrose,hard-leaved pocket orchid,canterbury bells,sweet pea,english marigold,tiger lily,moon orchid,bird of paradise,monkshood,globe thistle"

print(type(flowers))
print(flowers)

<class 'str'>
pink primrose,hard-leaved pocket orchid,canterbury bells,sweet pea,english marigold,tiger lily,moon orchid,bird of paradise,monkshood,globe thistle


Code di atas ini adalah contoh dari penggunaan variabel `flowers` yang diinisialisasi sebagai string panjang yang berisi nama-nama bunga yang dipisahkan oleh koma.
- `flowers` adalah sebuah variabel yang berisi string panjang yang terdiri dari nama-nama bunga yang dipisahkan oleh koma.
- `print(type(flowers))` mencetak tipe data dari variabel `flowers`. Di sini, variabel `flowers` adalah sebuah string.
- `print(flowers)` mencetak isi dari variabel `flowers`, yang merupakan daftar nama-nama bunga yang dipisahkan oleh koma.

In [128]:
flowers_list = ["pink primrose", "hard-leaved pocket orchid", "canterbury bells", "sweet pea", "english marigold", "tiger lily", "moon orchid", "bird of paradise", "monkshood", "globe thistle"]

print(type(flowers_list))
print(flowers_list)

<class 'list'>
['pink primrose', 'hard-leaved pocket orchid', 'canterbury bells', 'sweet pea', 'english marigold', 'tiger lily', 'moon orchid', 'bird of paradise', 'monkshood', 'globe thistle']


Code di atas ini menginisialisasi variabel `flowers_list` sebagai sebuah list yang berisi nama-nama bunga. Setiap nama bunga disimpan sebagai string dalam list tersebut.
- `flowers_list` adalah sebuah variabel yang berisi list yang berisi nama-nama bunga.
- `print(type(flowers_list))` mencetak tipe data dari variabel `flowers_list`. Di sini, variabel `flowers_list` adalah sebuah list.
- `print(flowers_list)` mencetak isi dari variabel `flowers_list`, yang merupakan daftar nama-nama bunga. Setiap nama bunga disimpan sebagai elemen terpisah dalam list tersebut.

# Length
Kita dapat menghitung jumlah entri dalam daftar mana pun dengan len(), yang merupakan kependekan dari "length" yang artinya panjang. Anda hanya perlu mencantumkan nama daftar di dalam tanda kurung.

In [129]:
# The list has ten entries
print(len(flowers_list))

10


Perintah len(flowers_list) digunakan untuk menghitung jumlah elemen dalam list flowers_list. Ini menunjukkan bahwa ada 10 nama bunga yang terdaftar dalam list flowers_list.

In [130]:
print("First entry:", flowers_list[0])
print("Second entry:", flowers_list[1])

# The list has length ten, so we refer to final entry with 9
print("Last entry:", flowers_list[9])

First entry: pink primrose
Second entry: hard-leaved pocket orchid
Last entry: globe thistle


Code di atas ini mengakses elemen-elemen tertentu dalam list `flowers_list` dan mencetaknya:
- `flowers_list[0]` mengakses elemen pertama dalam list `flowers_list` (indeks 0), yang merupakan "pink primrose".
- `flowers_list[1]` mengakses elemen kedua dalam list `flowers_list` (indeks 1), yang merupakan "hard-leaved pocket orchid".
- `flowers_list[9]` mengakses elemen terakhir dalam list `flowers_list` (indeks 9), karena indeks dimulai dari 0, sehingga elemen terakhir memiliki indeks 9. Ini adalah "globe thistle".

## Slicing

Anda juga dapat menarik segmen List (misalnya, tiga entri pertama atau dua entri terakhir). Ini disebut Slicing. Contohnya:
- untuk menarik entri x pertama, Anda menggunakan [:x], dan
- untuk menarik entri y terakhir, Anda menggunakan [-y:].

In [131]:
print("First three entries:", flowers_list[:3])
print("Final two entries:", flowers_list[-2:])

First three entries: ['pink primrose', 'hard-leaved pocket orchid', 'canterbury bells']
Final two entries: ['monkshood', 'globe thistle']


Code di atas ini mencetak subset tertentu dari list `flowers_list`:
- `flowers_list[:3]` menghasilkan subset dari list `flowers_list` mulai dari indeks 0 (inklusif) hingga indeks 3 (eksklusif), sehingga mencakup tiga elemen pertama. Ini akan mencetak tiga elemen pertama dari list tersebut: ["pink primrose", "hard-leaved pocket orchid", "canterbury bells"].
- `flowers_list[-2:]` menghasilkan subset dari list `flowers_list` mulai dari dua elemen terakhir (indeks -2) hingga elemen terakhir (indeks -1) secara inklusif. Ini akan mencetak dua elemen terakhir dari list tersebut: ["monkshood", "globe thistle"].

## Removing items

Hapus item dari daftar dengan .remove(), dan masukkan item yang ingin Anda hapus dalam tanda kurung.

In [132]:
flowers_list.remove("globe thistle")
print(flowers_list)

['pink primrose', 'hard-leaved pocket orchid', 'canterbury bells', 'sweet pea', 'english marigold', 'tiger lily', 'moon orchid', 'bird of paradise', 'monkshood']


Code flowers_list.remove("globe thistle") menghapus elemen "globe thistle" dari list flowers_list. Setelah perintah ini dijalankan, elemen tersebut tidak akan ada lagi dalam list.

## Adding items 

Tambahkan item ke daftar dengan .append(), dan masukkan item yang ingin Anda tambahkan dalam tanda kurung.


In [133]:
flowers_list.append("snapdragon")
print(flowers_list)

['pink primrose', 'hard-leaved pocket orchid', 'canterbury bells', 'sweet pea', 'english marigold', 'tiger lily', 'moon orchid', 'bird of paradise', 'monkshood', 'snapdragon']


Code flowers_list.append("snapdragon") menambahkan elemen baru dengan nilai "snapdragon" ke dalam list flowers_list

## Lists are not just for strings

Sejauh ini, kami hanya bekerja dengan daftar di mana setiap item dalam daftar adalah string. Namun daftar dapat memiliki item dengan tipe data apa pun, termasuk boolean, integer, dan float.

Sebagai contoh, perhatikan penjualan buku hardcover pada minggu pertama bulan April 2000 di toko retail.

In [134]:
hardcover_sales = [139, 128, 172, 139, 191, 168, 170]

Code di atas mendefinisikan sebuah list yang disebut hardcover_sales, yang berisi jumlah penjualan buku hardcover selama seminggu tertentu di toko retail. Setiap elemen dalam list ini mewakili jumlah buku hardcover yang terjual pada hari tertentu dalam urutan: Senin hingga Minggu.

In [135]:
print("Length of the list:", len(hardcover_sales))
print("Entry at index 2:", hardcover_sales[2])

Length of the list: 7
Entry at index 2: 172


Kode di atas ini mencetak panjang (jumlah elemen) dari list `hardcover_sales` dan mencetak elemen yang terletak pada indeks ke-2 dari list tersebut:
- `len(hardcover_sales)` mengembalikan jumlah elemen dalam list `hardcover_sales`, yang dalam kasus ini adalah jumlah penjualan selama seminggu, yaitu 7.
- `hardcover_sales[2]` mengakses elemen pada indeks ke-2 dalam list `hardcover_sales`, yang mewakili penjualan buku hardcover pada hari ketiga (indeks dimulai dari 0), yang nilainya adalah 172.

In [136]:
print("Minimum:", min(hardcover_sales))
print("Maximum:", max(hardcover_sales))

Minimum: 128
Maximum: 191


Kode di atas ini mencetak nilai minimum dan maksimum dari list `hardcover_sales`:
- `min(hardcover_sales)` mengembalikan nilai minimum dari list `hardcover_sales`, yaitu jumlah penjualan buku hardcover terendah dalam seminggu.
- `max(hardcover_sales)` mengembalikan nilai maksimum dari list `hardcover_sales`, yaitu jumlah penjualan buku hardcover tertinggi dalam seminggu.

Maka nilai minimum penjualan buku hardcover dalam seminggu adalah 128, sedangkan nilai maksimumnya adalah 191.

In [137]:
print("Total books sold in one week:", sum(hardcover_sales))

Total books sold in one week: 1107


Perintah `sum(hardcover_sales)` menghitung total penjualan buku hardcover selama satu minggu dengan menjumlahkan semua elemen dalam list `hardcover_sales`. Maka total buku yang terjual dalam satu minggu adalah 1107 buku hardcover.

In [138]:
print("Average books sold in first five days:", sum(hardcover_sales[:5])/5)

Average books sold in first five days: 153.8


Perintah di bawah ini menghitung rata-rata penjualan buku hardcover selama lima hari pertama (Senin hingga Jumat) dalam seminggu:
- `hardcover_sales[:5]` menghasilkan subset dari list `hardcover_sales` yang berisi jumlah penjualan buku hardcover pada lima hari pertama dalam seminggu.
- `sum(hardcover_sales[:5])` menghitung total penjualan buku hardcover selama lima hari pertama.
- Pembagian hasil jumlah penjualan dengan 5 menghitung rata-rata penjualan buku hardcover selama lima hari pertama.
Maka rata-rata penjualan buku hardcover selama lima hari pertama dalam seminggu adalah 153.8 buku.